types/document.d.ts

```diff
@@ -75,6 +75,7 @@ declare module 'mongoose' {
 
     /** Returns the model with the given name on this document's associated connection. */
     $model<ModelType = Model<unknown>>(name: string): ModelType;
+    $model<ModelType = Model<DocType>>(): ModelType;
```

Comment:
i dont see function `model` being available in the types, is this something that is forgotten or is there a reason it is not there?

Reply:
I don't know of any reason. My best guess would be to avoid issues back when we recommended using `& Document`, but don't see any reason why not. Looks like the tests pass with `model()` signature that matches `$model()`, so I will add that into this PR.